### PR TITLE
Check ceph usage

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**
to check by pool to support multi-end ceph cluster

### CODE REVIEW

File: sensu/plugins/check-ceph-usage.rb [hunk 1]
```diff
@@ -134,17 +134,25 @@ def run_cmd(cmd)
...
+      usage = pool['stats']['bytes_used'] * 100.0 /(pool['stats']['bytes_used'] + pool['stats']['max_avail'])
+      pool_usage[pool['name']] = usage
+      output = output + 'Pool ' + pool['name'] + " %Used: " + usage.to_s + "\n"
+    end unless result.to_s == ''
+    output = output + result if config[:verbose]
+
+    pool_usage.each do |pool_name, percentage|
+      if percentage >= config[:threshold]
+        warning output if config[:criticality] == 'warning'
+        critical output
```
Comment (omgjlk) [inline sensu/plugins/check-ceph-usage.rb]: Won't this report only the first pool with a critical level? What if multiple pools are in this state? Wouldn't it be better to report in aggregate, or leave this module as a single pool reporter and create multiple checks, one for each pool?

Comment (Xiaohua-Shen) [reply sensu/plugins/check-ceph-usage.rb]: all pools usage will be contained in the output, and any of these pools usage meet threshold will be reported as critical.

if both two pools meet critical threshold, output like this:
CheckCephClusterUsage Critical: Pool rbd_hybrid %Used: 81%
Pool rbd_ssd %Used: 76%

if the 2nd pool meet critical threshold, output like this:
CheckCephClusterUsage Critical: Pool rbd_hybrid %Used: 30%
Pool rbd_ssd %Used: 76%

I will modify the code with more meaningful output like check-disk

**Comment by Xiaohua-Shen** [comment]
code is changed, the sample output is like below：

```
root@cpm1:/etc/sensu/plugins# ./check-ceph-usage.rb -w 75
CheckCephPoolUsage CRITICAL: rbd_hybrid 76%, rbd_ssd 86%
root@cpm1:/etc/sensu/plugins# ./check-ceph-usage.rb -w 80
CheckCephPoolUsage CRITICAL: rbd_ssd 86%
root@cpm1:/etc/sensu/plugins# ./check-ceph-usage.rb -w 90
CheckCephPoolUsage OK: All Pools usage under 90%
root@cpm1:/etc/sensu/plugins# ./check-ceph-usage.rb -w 75 --criticality warning
CheckCephPoolUsage WARNING: rbd_hybrid 76%, rbd_ssd 86%
```

**Comment by omgjlk** [comment]
Hi @Xiaohua-Shen . Thanks for this, but after some discussion with other consumers of ursula-monitoring, we'd prefer this go in a different direction. We'd rather this plugin grow an argument for which pool to look at (maybe with a reasonable default) so that we can create multiple checks with different configurations that ultimately use the same plugin. The plugin would continue to focus itself on a single pool, either the reasonable default or the specific pool provided to it in an argument.

Does this make sense?

**Review by omgjlk — CHANGES_REQUESTED** [review]
Please redesign this per comments.

**Comment by Xiaohua-Shen** [comment]
the pool names are not hardcoded in the check code. it's based on the `ceph df --format=json` output, it will contain all the pools existed in this ceph cluster. 
I can add one more optional parameter as pools for the check. if this parameter is provided, then only specific pools will be checked, otherwise all pools will be checked.

```
# ./check-ceph-usage.rb -w 0 -p default
CheckCephPoolUsage CRITICAL: default 0.23130386373200215%
# ./check-ceph-usage.rb -w 0
CheckCephPoolUsage CRITICAL: default 0.23130386373200215%
# ./check-ceph-usage.rb -w 0 -p aaa
CheckCephPoolUsage OK: All Pools usage under 0%
# ./check-ceph-usage.rb -w 0 -p aaa,default
CheckCephPoolUsage CRITICAL: default 0.23130400192492817%
# ceph df
GLOBAL:
    SIZE       AVAIL      RAW USED     %RAW USED
    67026G     66870G         155G          0.23
POOLS:
    NAME        ID     USED       %USED     MAX AVAIL     OBJECTS
    default     1      52890M      0.23        22278G       12901
```

**Comment by knandya** [comment]
@j2sol Does the latest change set look ok?

**Review by omgjlk — APPROVED** [review]

**Pull request merged**